packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -276,6 +277,66 @@ gmaps.Icon? _gmIconFromBitmapDescriptor(BitmapDescriptor bitmapDescriptor) {
           ..size = size
           ..scaledSize = size;
       }
+    } else if (iconConfig[0] == 'asset') {
+      assert(iconConfig.length >= 2);
+      // iconConfig[2] contains the DPIs of the screen, but that information is
+      // already encoded in the iconConfig[1]
+
+      final String assetUrl =
+          ui.webOnlyAssetManager.getAssetUrl(iconConfig[1]! as String);
+      icon = gmaps.Icon()..url = assetUrl;
+
+      if (iconConfig.length == 3) {
+        final double scale = iconConfig[2]! as double;
+        // Google Maps Web SDK does not support the scaling of the marker with anything other than
+        // the size parameter, therefore the width and height of the image must be read from the image.
+        // To avoid this, it is best to provide the image size instead of scale when using the web platform.
+        final ui.Codec codec =
+            await ui.webOnlyInstantiateImageCodecFromUrl(Uri.parse(assetUrl));
```

Comment:
I like this new approach, but I don't think this works in the `html` renderer?

https://github.com/flutter/engine/blob/9080a2b49e8cd385c254dd3cdd1685e26b8708a9/lib/web_ui/lib/src/engine/html/renderer.dart#L184

Reply:
True, and this also decodes the entire image frame for each marker, as there is no caching either, which is not a really efficient solution, but not too slow either. I like this solution as this allows automatically use exact same logical marker size for each platform, including web. But this is indeed a bit slow compared to using size parameter (which is fastest approach for web), and also not working for `html` renderer
I would like to keep this support for scaling with imagePixelRatio, but change the logic so that image width and height are parsed from the image header, so that only first bytes of data need to be parsed... and check that this approach is fast enough and valid for `html` renderer as well. [Here's an example for this approach by @dnfield](https://gist.github.com/dnfield/ed17cfc37a1f8fabad13431329328246). Writing something similar as this gist has no proper license to be used directly.
And also API need to be changed a bit, as developers are not able to disable scaling for web at the moment as for web images need to be _counter-scaled_ even if imagePixelFactor is 1.

Reply:
@jokerttu, have you see `package:image`? The only concern I have with this approach is that you'll need to download with a XHR request the image to get access to its bytes, and that's going to make this slower. Is there any way of combining the two solutions, so users can pass a size and dpr for a given marker, and if not, fall back to the slower code path?

Reply:
@ditman yes, image packages [startDecode] (https://pub.dev/documentation/image/latest/image/Decoder/startDecode.html) does exactly this. If it is ok to add new dependency this would be better option than writing own parsers for image headers.
`webOnlyInstantiateImageCodecFromUrl` fetches image as well, for each marker, but of course image request is cached by browser, but it still fills up network panel with requests each time marker updates. This is not really good thing either.
I might still implement this feature, and keep the feature for web as well, and instruct users to use size instead, as more performant option.

Reply:
@ditman feature implemented using `package:image`. Also fixed case where image was scaled even when imageScaling was set to noScaling.
On web, using scaling without "size" is slow, as the image need to be loaded before size can be calculated, but users have way to improve performance by using size or disabling scaling totally.